android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -225,14 +225,21 @@ private void performAuthorization(AuthorizationServiceConfiguration serviceConfi
             authRequestBuilder.setAdditionalParameters(additionalParameters);
         }
 
-        AppAuthConfiguration.Builder authConfigBuilder = new AppAuthConfiguration.Builder();
-        if (allowInsecureConnections) {
-            authConfigBuilder.setConnectionBuilder(InsecureConnectionBuilder.INSTANCE);
-        }
+        // Launch the AppAuth on a separate thread so that this one can immediately return
+        // to Flutter, preventing the UI from being locked up.
+        Thread t = new Thread(new Runnable() {
```

Comment:
Thanks for the PR. From my understanding, short tasks can use `AsyncTask`, which also makes it easier to return results on the UI thread (which is how I've seen done in other plugins). `google_sign_in` plugin has their own class for executing work in the background that then returns results in the UI thread but `AsyncTask` should suffice.  If this works (note: need to ensure cancelling propagates the exception to the Flutter side) then I reckon the changes could be applied where needed. I haven't taken a look at the other methods but it's possible that only applying it to `performAuthorization`  is sufficient

Reply:
Thanks for the insight! I'll preface with I'm no expert in Android, so this is a good experience for me. One point- we technically don't ever return anything to Flutter from the method call since everything is done asynchronously via the redirect URL / deep link. Do we need to go that deep into it?

Is this simple thread approach not recommended do you think?

Reply:
Another data point - with my testing, iOS seems to handle it fine with no UI blocking in the current implementation.

Reply:
Assuming the token call needs the change, that is one that returns a result. Also, the authorisation related methods start an activity and that's a UI related activity so should happen on the main thread. AsyncTask has a way to do this. Everything that takes place before then could happen in the background. Might be worth digging deeper to see why that causes a performance issue. I suspect it's to do with fetching with configuration info.